cmd/xc/handler/handlers.go

```diff
@@ -0,0 +1,224 @@
+package handler
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	xc "github.com/cordialsys/crosschain"
+	"github.com/cordialsys/crosschain/factory"
+
+	"github.com/sirupsen/logrus"
+)
+
+func BalanceHandler(contract string, xcFactory *factory.Factory, chainConfig *xc.ChainConfig, args []string) (string, error) {
```

Comment:
I think these handlers aren't really worth it.  It's a lot of fluff just to ultimately do `cli.FetchBalance(...)` which is what we want to test.

Rather than make new abstractions I'd prefer to just use crosschain methods directly in the tests.  Not only will this be easier to follow, but I think it'll make for less code as well.

For example, all of these "handlers" they repeat code, like `xcFactory.NewClient` and `xcFactory.MustAddress`.  You're also still passing vague things like `arg []string` and it's not clear what the concrete types are.

Also passing `args []string` is still problematic

Reply:
Reworked it:
- removed handlers and made specific methods for common parts used by both command and tests so we won't have code duplicates
- managed every repetition we have in the code in terms as you've pointed above
- fix passing vague things and only use concrete types